### ISSUE TITLE
Filter activities from external apps in activity stream

### ISSUE THREAD

**Issue by marius-wieschollek** [nextcloud/activity#44]
The real life example I have is a webcam that uploads a photo every few minutes via webdav. Of course all the photos show up in the activity stream and make other activities hard to find. Recently Nextcloud has introduced the great app passwords feature for external apps. Since said webcam uses its own app password to upload files,  I thought it may be possible to exclude if from the stream by its app password or the like.

### Steps to reproduce
1. Open "Activity" App
2. Select "All Activities"

### Expected behaviour
Activity offers a way to exclude activities created by an external app from the stream by its app password.

### Actual behaviour
Activity shows unfiltered stream with lots of automatically generated activities from external apps.

### Server configuration
**Operating system**: Ubuntu 14.04
**Web server:**: Apache 2.4.23
**Database:** mysql 5.6.31
**PHP version:** PHP 7.0.11
**Nextcloud version:** 10.0
**Where did you install Nextcloud from:** Website Zip-download
**Signing status:** disabled

**Comment by nickvergessen** [comment]
Hmm, I see the problem, but this is nothing the activity app can take care of, since the API doesn't provide the authentication, but just the user id.
when it is the same folder for you all the time, you can add a check for the folder, and hack in to skip that.
Paste me some file+folder names and I will write you the changes.

**Comment by marius-wieschollek** [comment]
@nickvergessen Thanks for your answer. Actually I had added some code to \OCA\Activity\FilesHooks::fileCreate() to exclude the folder as a temporary solution. But I ran into some problems when I tried to move that code to an extra app. Since this issue is not the right place to discuss these I have opened a topic in the Nextcloud forums. (https://help.nextcloud.com/t/override-other-classes-with-my-app/3419)

Thank you very much for your help and thank you for your offer.
If you don't mind I would try to see if I can get this working myself as an nextcloud app.
